zookeeper-client/zookeeper-client-c/src/zookeeper.c

```diff
@@ -2592,11 +2592,6 @@ static int init_ssl_for_socket(zsock_t *fd, zhandle_t *zh, int fail_on_error) {
         OPENSSL_init_ssl(OPENSSL_INIT_LOAD_SSL_STRINGS | OPENSSL_INIT_LOAD_CRYPTO_STRINGS, NULL);
         method = TLS_client_method();
 #endif
-        if (FIPS_mode() == 0) {
```

Comment:
nit : just a suggestion - code clean up can be done separately from the actual code change. j

Reply:
This commit is/change required to unblock CI builds. See https://github.com/apache/zookeeper/pull/1951 .